test/torchaudio_unittest/datasets/librimix_test.py

```diff
@@ -0,0 +1,100 @@
+
+_SAMPLE_RATE = 8000
+_TASKS_TO_MIXTURE = {
+    "sep_clean": "mix_clean",
+    "enh_single": "mix_single",
+    "enh_both": "mix_both",
+    "sep_noisy": "mix_both",
+}
+
+
+def _save_wav(filepath: str, seed: int):
+    wav = get_whitenoise(
+        sample_rate=_SAMPLE_RATE,
+        duration=0.01,
+        n_channels=1,
+        seed=seed,
+    )
+    save_wav(filepath, wav, _SAMPLE_RATE)
+    return wav
+
+
+def get_mock_dataset(root_dir: str, num_speaker: int):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    mocked_data = []
+    seed = 0
+    base_dir = os.path.join(root_dir, f"Libri{num_speaker}Mix", "wav8k", "min", "train-360")
+    os.makedirs(base_dir, exist_ok=True)
+    for utterance_id in range(10):
+        filename = f"{utterance_id}.wav"
+        task_outputs = {}
+        for task in _TASKS_TO_MIXTURE:
+            # create mixture folder. The folder names depends on the task.
+            mixture_folder = _TASKS_TO_MIXTURE[task]
+            mixture_dir = os.path.join(base_dir, mixture_folder)
+            os.makedirs(mixture_dir, exist_ok=True)
+            mixture_path = os.path.join(mixture_dir, filename)
+            mixture = _save_wav(mixture_path, seed)
+            sources = []
+            if task == "enh_both":
+                sources = [task_outputs["sep_clean"][1]]
+            else:
+                for speaker_id in range(num_speaker):
+                    source_dir = os.path.join(base_dir, f"s{speaker_id+1}")
+                    os.makedirs(source_dir, exist_ok=True)
+                    source_path = os.path.join(source_dir, filename)
+                    if os.path.exists(source_path):
+                        sources = task_outputs["sep_clean"][2]
+                        break
+                    else:
+                        source = _save_wav(source_path, seed)
+                        sources.append(source)
+                        seed += 1
+            task_outputs[task] = (_SAMPLE_RATE, mixture, sources)
+        mocked_data.append(task_outputs)
+
+    return mocked_data
+
+
+class TestLibriMix(TempDirMixin, TorchaudioTestCase):
+    backend = "default"
+
+    root_dir = None
+    samples_2spk = []
+    samples_3spk = []
+
+    @classmethod
+    def setUpClass(cls):
+        cls.root_dir = cls.get_base_temp_dir()
+        cls.samples_2spk = get_mock_dataset(cls.root_dir, 2)
+        cls.samples_3spk = get_mock_dataset(cls.root_dir, 3)
+
+    def _test_librimix(self, dataset, samples, task):
+        num_samples = 0
+        for i, (sample_rate, mixture, sources) in enumerate(dataset):
+            assert sample_rate == samples[i][task][0]
+            self.assertEqual(mixture, samples[i][task][1], atol=5e-5, rtol=1e-8)
+            assert len(sources) == len(samples[i][task][2])
+            for j in range(len(sources)):
+                self.assertEqual(sources[j], samples[i][task][2][j], atol=5e-5, rtol=1e-8)
```

Comment:
is there a reason why default atol/rtol values dont work?

Reply:
not really, I was implementing it based on `librispeech_test`. 
https://github.com/pytorch/audio/blob/0b4b1fd4296fd7d54f0059a23cf9c9e168ad9376/test/torchaudio_unittest/datasets/librispeech_test_impl.py#L70
Removing it also passes the test. Will update the pr.